lib/editor-linter.coffee

```diff
@@ -1,102 +1,42 @@
-{CompositeDisposable, Emitter, Range} = require 'atom'
-Helpers = require './helpers'
+{TextEditor, Emitter, CompositeDisposable} = require('atom')
 
 class EditorLinter
-  constructor: (@linter, @editor) ->
-    @status = true
-    @messages = new Map
-    @inProgress = false
-    @inProgressFly = false
-
-    if @editor is atom.workspace.getActiveTextEditor()
-      @linter.views.updateLineMessages(true)
-
+  constructor: (@editor) ->
+    throw new Error("Given editor isn't really an editor") unless @editor instanceof TextEditor
+    @status = true # Overall linting status
     @emitter = new Emitter
     @subscriptions = new CompositeDisposable
-
-    @subscriptions.add(
-      @editor.onDidSave => @lint(false)
-    )
-    @subscriptions.add(
-      @editor.onDidChangeCursorPosition ({oldBufferPosition, newBufferPosition}) =>
+    @subscriptions.add @editor.onDidDestroy =>
+      @emitter.emit 'did-destroy'
+
+    @subscriptions.add @editor.onDidSave => @emitter.emit('should-lint', false)
+    # The onDidStopChanging callbacks are invoked immediately, and we want to avoid it
+    setImmediate =>
+      @subscriptions.add @editor.onDidStopChanging => setImmediate => @emitter.emit('should-lint', true)
```

Comment:
No, I've defined `emitShouldLint`, when called it will asynchronously trigger `should-lint`.

Now that we have an asynchronous emitter we can register that callback. The only difference between my code and yours is that I named the function instead of nesting 2 anonymous functions inside of a callback registration